chips/rp2040/src/pio.rs

```diff
@@ -506,6 +515,55 @@ const PIO1_SET_BASE: StaticRef<PioRegisters> =
 const PIO1_CLEAR_BASE: StaticRef<PioRegisters> =
     unsafe { StaticRef::new((PIO_1_BASE_ADDRESS + 0x3000) as *const PioRegisters) };
 
+/// Represents a relocated PIO program.
+///
+/// An [Iterator] that yields the original program except `JMP` instructions have
+/// relocated target addresses based on an offset.
```

Comment:
```suggestion
/// relocated target addresses based on an offset.
///
/// This follows the PIO Assembler reference in Section 3.3 of the Raspberry Pi
/// RP2040 Datasheet [1].
///
/// [1]: https://datasheets.raspberrypi.com/rp2040/rp2040-datasheet.pdf
```

To implement @alexandruradovici's suggestion of https://github.com/tock/tock/pull/4218#discussion_r1824428806.

Reply:
Maybe better include link to PIO documentation in the beginning of that file?